test/module/irohad/multi_sig_transactions/gossip_propagation_strategy_test.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ametsuchi/peer_query.hpp"
+#include "multi_sig_transactions/gossip_propagation_strategy.hpp"
+#include <algorithm>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+#include <model/peer.hpp>
+#include <rxcpp/rx.hpp>
+#include <string>
+#include <vector>
+
+namespace iroha {
+
+using namespace std::chrono_literals;
+using PropagationData = GossipPropagationStrategy::PropagationData;
+
+class MockPeerQuery : public ametsuchi::PeerQuery {
+public:
+  MOCK_METHOD0(getLedgerPeers, nonstd::optional<PropagationData>());
+};
+
+/**
+ * @given list of peers and
+ *        strategy that emits two peers in some time
```

Comment:
It looks like strategy emits not 2 peers at time but half of peers (or may be I misunderstand commentary).

Reply:
it emits 2 peers (look at the `amount`)

call `take(n)` actually fetch n times this pairs so -> `peers.size() / amount * amount = peers.size()` if it's even